tabletop/core/game.js

```diff
@@ -293,4 +307,186 @@ Game.prototype.destroyToken = function(token) {
+
+      var gameCopy = this.copyGameStatus(this);
+
+      var tokenCp = gameCopy.board.tokens[tokenIdx];
+      var tileCp, destCp;
+      if (game.board instanceof Gridboard) { 
+        tileCp = gameCopy.board.tiles[tilePos.x][tilePos.y];
+        destCp = gameCopy.board.tiles[destPos.x][destPos.y];
+      } else { 
+        tileCp = gameCopy.board.tiles[tilePos];
+        destCp = gameCopy.board.tiles[destPos];
+      } 
+        
+      gameCopy.proposedMove = {
+        token: tokenCp,
+        tile: tileCp,
+        destination: destCp,
+        tokenIdx: tokenIdx, 
+        tilePos: tilePos, 
+        destPos: destPos
+      };
+      
+      gameCopy.executeMove();
+      results.push({
+        score: gameCopy.scoreBoard(player), 
+        token: tokenCp, 
+        tile: tileCp, 
+        destination: destCp,
+        tokenIdx: tokenIdx, 
+        tilePos: tilePos, 
+        destPos: destPos
+      });
+    }, this);
+  }
+  
+  var bestResult = results[0];
+  var bestScore = 0;
+  
+  results.forEach(function(result) { 
+    if (result.score > bestScore) { 
+      bestResult = result; 
+      bestScore = result.score;
+    }
+  });
+
+  
+  var actualToken = this.board.tokens[bestResult.tokenIdx];
+  // actualTile should be same as this.board.tile[tilePos], but we 
+  // use this cause it generalizes to all board types 
+  var actualTile = this.board.findTileForToken(actualToken);
+
+  var actualDest;
+  if (this.board instanceof Gridboard) { 
+    actualDest = this.board.getTile(bestResult.destPos.x, bestResult.destPos.y);
+  } else { 
+    actualDest = this.board.getTile(bestResult.destPos);
+  } 
+  
+  var actualBestMove = 
+        { 
+          token: actualToken,
+          tile: actualTile,
+          destination: actualDest
+        };
+  
+  return actualBestMove;
+};
+
+Game.prototype.getValidMoves = function(player) { 
+
+  var legalMoves = {};
+  legalMoves.tokens = [];
+  legalMoves.tiles = [];
+  legalMoves.destinationTiles = [];
+  if (this.moveType == c.moveTypeManual) { 
+    
+    player.tokens.forEach(function(token) { 
+      var tile = this.board.findTileForToken(token);
+      var destinationTiles = [];
+      // since this.board is a double array we need to double loop
+      this.board.tiles.forEach(function(destinationRow) { 
+        destinationRow.forEach(function(destination) { 
+          if (this.isValidMove(token, tile, destination))
+            destinationTiles.push(destination);
+        }, this);
+      }, this);
+      
+      if (destinationTiles.length > 0) { 
+        legalMoves.tokens.push(token);
+        legalMoves.tiles.push(tile);
+        legalMoves.destinationTiles.push(destinationTiles);
+      }
+    }, this);
+    
+  } 
+  
+  return legalMoves;
+};
+
+// Create copy of game and manually copy over 
+// tiles/tokens to remove references
+Game.prototype.copyGameStatus = function(game) {
+
+  //var newGame = $.extend(true, {}, game);
+  var newGame = Object.create(this);
```

Comment:
@andrewjmeier Object.create() here copies the attributes of the game over to newGame